requirements/specifications/lifecycle/index.md

```diff
@@ -0,0 +1,675 @@
+the Lifecycle.onInactive notification with the current state and
+previous states.
+
+Apps in this state **MUST** not use the media pipeline and the platform
+**MUST** tear down any Media Pipeline sessions associated with this app.
+
+When an app is initializing, and registers for Lifecycle.onInactive
+notifications, the platform **MUST** transition the app to the inactive
+state with the previous state set to initializing, since it is now able
+to execute any code related to that transition.
+
+If an app is still initializing, and calls Lifecycle.ready(), the
+platform **MUST** transition the app to the inactive state with the
+previous state set to initializing.
+
+When an app is in either *active* state and is *closed* then the
+platform **MUST** transition the app to the inactive state with the
+previous and current state set correctly.
+
+When an app is in the suspended state and is then unsuspended, then the
+platform **MUST** transition the app to the inactive state with the
+previous state set to suspended.
+
+### 2.3. Foreground
+
+This state allows an app to be presented as the primary experience to
+the user.
+
+The platform **MUST** ensure that only one or zero apps is ever in this
+state at one time.
+
+Apps in this state **MUST** be visible.
+
+Apps in this state **MUST** receive RCU key presses that they have
+permissions for.
+
+Apps in this state **MUST** have access to the audio-video decoder.
+
+Apps in this state **SHOULD NOT** have performance negatively impacted
+by other processes on the device.
+
+When an app transitions to this state, the platform **MUST** dispatch
+the Lifecycle.onForeground notification with the current state and
+previous states.
+
+### 2.4. Background
+
+This state allows an app to be present as the secondary experience to
+the user, e.g. when a system settings UI is overlayed on top of the app.
+
+Different platforms **MAY** support differing numbers of background
+apps. See [Background App
+Requirements](https://comcastcorp.sharepoint.com/:w:/r/sites/hqppa/Shared%20Documents/Global%20App%20Platform/Firebolt/Requirements/Core/Lifecycle/Background%20App%20Requirements.docx?d=w185d9112bcd44376bb0635dad816919b&csf=1&web=1&e=cjmuSJ)
+for more info.
+
+Apps in this state **MUST** either be partially visible **OR** have
+access to an audio decoder, e.g. an app playing music in the background.
+
+Apps in this state **SHOULD NOT** receive most RCU key presses, although
+platforms **MAY** forward certain keys to the app, e.g. play/pause keys.
+
+Apps in this state **MAY** have access to the audio-video decoder. How
+many decoders are available is a decision for each device based on
+resources. See [Background App
+Requirements](https://comcastcorp.sharepoint.com/:w:/r/sites/hqppa/Shared%20Documents/Global%20App%20Platform/Firebolt/Requirements/Core/Lifecycle/Background%20App%20Requirements.docx?d=w185d9112bcd44376bb0635dad816919b&csf=1&web=1&e=cjmuSJ)
+for more info.
```

Comment:
There shall be some use cases discussed regarding audio-video decoder management between the foreground and background apps.
The following cases shall be discussed:
1. System ui overlying app that uses audio-video decoder.
2. Audio player in background (not visible)

How the resource conflicts between foreground and background apps shall be solved:
- shall the foreground app always be prioritized over background app when allocating AV decoders.
- shall the audio playback from background app be stopped when foreground app starts audio playback?
- then when foreground app stops av playback, shall the background playback be resumed?